src/strkey.rs

```diff
@@ -122,7 +75,10 @@ fn decode(s: &str) -> Result<(Version, Vec<u8>), DecodeError> {
         if data.len() < 3 {
             return Err(DecodeError::Invalid);
         }
-        let ver = Version::try_from(data[0])?;
+        let ver = match Version::try_from(data[0]) {
+            Ok(ver) => ver,
+            Err(_) => return Err(DecodeError::Invalid),
+        };
```

Comment:
💡 Recommend using `.map_err` instead of the match, since you only need to map the error and not the value. It's a little less verbose than what is here right now, but same outcome.

Alternatively we could `impl From<TryFromPrimitiveError<Version>> for DecodeError` and map the error directly to `DecodeError::Invalid`. If you do that, you can keep the `?` and it will be automatically converted to the appropriate type. This has the advantage of meaning anywhere this error occurs will be automatically wrapped, which is probably good for this specific error.